src/main/kotlin/com/team4099/robot2023/subsystems/superstructure/Superstructure.kt

```diff
@@ -0,0 +1,990 @@
+              else -> 0.0.inches
+            }
+          manipulator.currentRequest =
+            Request.ManipulatorRequest.TargetingPosition(
+              extension, ManipulatorConstants.IDLE_VOLTAGE
+            )
+          elevator.currentRequest = Request.ElevatorRequest.TargetingPosition(scoreHeight)
+        }
+
+        // Transition
+        if (currentRequest is SuperstructureRequest.Score &&
+          groundIntake.isAtTargetedPosition &&
+          manipulator.isAtTargetedPosition &&
+          elevator.isAtTargetedPosition
+        ) {
+          nextState =
+            when (usingGamePiece) {
+              GamePiece.CUBE -> SuperstructureStates.SCORE_CUBE
+              GamePiece.CONE -> SuperstructureStates.SCORE_CONE
+              else -> SuperstructureStates.IDLE
+            }
+        } else if (currentRequest is SuperstructureRequest.Idle) {
+          nextState = SuperstructureStates.IDLE
+        }
+      }
+      SuperstructureStates.SCORE_CUBE -> {
+        // Outputs
+        manipulator.currentRequest =
+          Request.ManipulatorRequest.TargetingPosition(
+            manipulator.armPositionTarget,
+            Manipulator.TunableManipulatorStates.cubeOutVoltage.get()
+          )
+
+        // Transition
+        if (manipulator.isAtTargetedPosition &&
+          (Clock.fpgaTime - lastTransitionTime) >=
+          Manipulator.TunableManipulatorStates.spitTime.get()
+        ) {
+          nextState = SuperstructureStates.IDLE
+        }
+      }
+      SuperstructureStates.SCORE_CONE -> {
+        // Outputs
+        val dropToPosition =
+          when (nodeTier) {
+            NodeTier.HYBRID -> Elevator.TunableElevatorHeights.hybridHeight.get()
+            NodeTier.MID -> Elevator.TunableElevatorHeights.midConeHeight.get()
+            NodeTier.HIGH -> Elevator.TunableElevatorHeights.highConeHeight.get()
+            else -> elevator.elevatorPositionTarget
+          }
+        elevator.currentRequest = Request.ElevatorRequest.TargetingPosition(dropToPosition)
+        if (elevator.isAtTargetedPosition) {
+          manipulator.currentRequest =
+            Request.ManipulatorRequest.TargetingPosition(
+              manipulator.armPositionTarget,
+              Manipulator.TunableManipulatorStates.coneOutVoltage.get()
+            )
+        }
+
+        // Transition
+        if (elevator.isAtTargetedPosition &&
+          manipulator.isAtTargetedPosition &&
+          (Clock.fpgaTime - lastTransitionTime) >=
+          Manipulator.TunableManipulatorStates.spitTime.get()
+        ) {
+          nextState = SuperstructureStates.IDLE
+          currentRequest = SuperstructureRequest.Idle()
+        }
+      }
+      SuperstructureStates.TUNING -> {}
+    }
+
+    if (nextState != currentState) {
+      lastTransitionTime = Clock.fpgaTime
+    }
+
+    currentState = nextState
+  }
+
+  // Superstructure Commands
+  fun prepscoreConeHighCommand(): CommandBase {
+    val returnCommand = runOnce {
```

Comment:
can we do an `until` here or something, so that the command will end once the superstructure has achieved this? could just check the state for this. This would allow you to sequence with a `score` command or know when the auto can continue or something like this